src/state_manager/mod.rs

```diff
@@ -510,14 +510,20 @@ where
         tipset: &Arc<Tipset>,
     ) -> anyhow::Result<StateOutput> {
         let key = tipset.key();
-        self.cache
-            .get_or_else(key, || async move {
+        Ok(match self.cache.get(key) {
```

Comment:
@lemmih this should be sufficient for fixing RPC methods, but do you think this should happen in `populate_cache` instead?